src/main/scala/cromwell/binding/RuntimeAttributes.scala

```diff
@@ -2,126 +2,217 @@ package cromwell.binding
+  }
+
+  private def validateCpu(value: Option[String]): ValidationNel[String, RuntimeCpu] = {
+    value match {
+      case Some(v) => validateLong(v) map RuntimeCpu
+      case None => Defaults.Cpu.successNel
     }
+  }
 
-    def processRuntimeAttributes(astList: AstList): Map[String, String] = {
-      astList.asScala.toVector map { a => processRuntimeAttribute(a.asInstanceOf[Ast]) } toMap
+  private def validatePreemptible(value: Option[String]): ValidationNel[String, RuntimePreemptible] = {
+    value match {
+      case Some(v) => validateBoolean(v) map RuntimePreemptible
+      case None => Defaults.Preemptible.successNel
     }
+  }
 
-    val asts = ast.findAsts(AstNodeName.Runtime)
-    if (asts.size > 1) throw new UnsupportedOperationException("Only one runtime block may be defined per task")
-    val astList = asts.headOption map {
-      _.getAttribute("map").asInstanceOf[AstList]
+  private def validateFailOnStderr(value: Option[String]): ValidationNel[String, RuntimeFailOnStderr] = {
+    value match {
+      case Some(v) => validateBoolean(v) map RuntimeFailOnStderr
+      case None => Defaults.FailOnStderr.successNel
     }
-    val attributeMap = astList map processRuntimeAttributes getOrElse Map.empty[String, String]
-    val actualKeys = attributeMap.keySet
-    val knownKeys = RuntimeKey.values().map { _.key }
+  }
 
-    // Finding keys unknown to any backend is an error.
-    val unknownKeys = actualKeys -- knownKeys
-    if (unknownKeys.nonEmpty) {
-      throw new RuntimeException(s"Unknown keys found in runtime configuration: "
-        + unknownKeys.toSeq.sorted.mkString(", "))
+  private def validateLocalDisks(value: Option[String]): ValidationNel[String, RuntimeDisks] = {
+    value match {
+      case Some(v) =>
+        val nels = v.split(",\\s*").toList map { x => validateLocalDisk(x) }
+        // Use the LocalizationDisk as the start of the fold. Currently same as default, but perhaps not always
+        val defaultDiskNel = RuntimeDisks(Vector(LocalizationDisk)).successNel[String]
+        nels.foldLeft(defaultDiskNel)((acc, v) => (acc |@| v) { (a, v) => RuntimeDisks(a.disks :+ v) })
+      case None => Defaults.Disk.successNel
+    }
+  }
+
+  private def validateLocalDisk(disk: String): ValidationNel[String, Disk] = {
+    try {
+      val Array(name, sizeGb, diskType) = disk.split("\\s+")
+      validateLong(sizeGb) map { LocalDisk(name, _, diskType).toDisk }
+    } catch {
+      case _: MatchError => s"$disk should be in form NAME SIZE TYPE with SIZE in GB".failureNel
+    }
+  }
+
+  private def validateMemory(value: Option[String]): ValidationNel[String, RuntimeMemory] = {
+      value match {
+        case Some(v) => validateMemStringInGb(v) map RuntimeMemory
+        case None => Defaults.Memory.successNel
+      }
+  }
+
+  private def validateMemStringInGb(mem: String): ValidationNel[String, Double] = {
+    val amountAndUnit = mem.split("\\s+")
```

Comment:
This could be tolerant of non-whitespace with a `"(\\d+)\\s*(\\w)"` pattern match instead of a `split`